context/context.go

```diff
@@ -360,7 +360,26 @@ func (context *AptlyContext) PackagePool() aptly.PackagePool {
 	defer context.Unlock()
 
 	if context.packagePool == nil {
-		context.packagePool = files.NewPackagePool(context.config().RootDir, !context.config().SkipLegacyPool)
+		storageConfig := context.config().PackagePoolStorage
+		if storageConfig.Azure != nil {
+			var err error
+			context.packagePool, err = azure.NewPackagePool(
+				storageConfig.Azure.AccountName,
+				storageConfig.Azure.AccountKey,
+				storageConfig.Azure.Container,
+				storageConfig.Azure.Prefix,
+				storageConfig.Azure.Endpoint)
+			if err != nil {
+				Fatal(err)
+			}
+		} else {
+			poolRoot := context.config().PackagePoolStorage.Path
+			if poolRoot == "" {
+				poolRoot = filepath.Join(context.config().RootDir, "pool")
+			}
+
+			context.packagePool = files.NewPackagePool(poolRoot, !context.config().SkipLegacyPool)
+		}
 	}
```

Comment:
Can we make the package pool storage configs similar to publish configs and let user make the choice instead of preferential based selection here? For example, the users can have `FileSystemPackagePoolRoots` and `AzurePackagePoolRoots` with settings similar to `FileSystemPublishRoots` and `AzurePublishRoots`. The `PackagePoolStorage` will select an existing config. I think this will the config less confusing to the users.

Reply:
I will admit, I'm kind of mixed on the current configuration approach :sweat_smile: My rationale was two-fold:

- It shouldn't look like it's easy / workable to randomly change package pools, once it's set to filesystem or azure, it's basically *set*, unlike publishing where you can decide to just publish somewhere else
- Having parallel `AzurePackagePoolRoots` and `AzurePublishRoots`, both probably containing identical key-value pairs, feels a bit ugly

So I was split between two solutions:

- Rename the `*PublishRoots` to, say, `*Endpoints`, and have the `PackagePoolStorage` reference that
- The current solution

and I just went over the current approach due to the first point, but again I'm not really a huge fan / not tied to it. Would going with the other solution also work? (Presumably backwards-compat would be kept, of course! Having the configuration split between two likely near-identical places just feels off IMO...even though that's sort of the current solution anyway I guess?)

Reply:
I understand your concern. The way I read the current solution with multiple choices is that both the local path and remote will be used as package pool storage instead of picking one over another. Probably we should either err on more than one config or actually store the packages on all configured storage pools. What do you think?

Reply:
I think either `type` or `AzureEndpoints` works. 

>  even though you can't switch package pools right now
Can I upload an existing local package pool to the Azure blob storage and set the pool config to the storage account to make the switch?

Reply:
> Can I upload an existing local package pool to the Azure blob storage and set the pool config to the storage account to make the switch?

ATM the paths are different, but I can make them consistent with the normal package pool and then it should work. (Technically, the legacy path locations won't work, but I'm not sure if that's something worth adding.)

Reply:
I don't think we need to support the legacy path. But making both package pool paths consistent is definitely a huge plus, i.e. users can just upload the existing local package pool to Azure and update the config to switch to the remote package pool.